cuda_bindings/tests/test_nvjitlink.py

```diff
@@ -53,71 +76,82 @@ def test_invalid_arch_error():
-    nvjitlink.add_file(handle, nvjitlink.InputType.ANY, str(file_path))
-    nvjitlink.complete(handle)
-    nvjitlink.destroy(handle)
+    for option, ptx_bytes in zip(ARCHITECTURES, ptx_kernel_bytes):
+        handle = nvjitlink.create(1, [f"-arch={option}"])
+        file_path = tmp_path / "test_file.cubin"
+        file_path.write_bytes(ptx_bytes)
+        nvjitlink.add_file(handle, nvjitlink.InputType.ANY, str(file_path))
+        nvjitlink.complete(handle)
+        nvjitlink.destroy(handle)
 
 
 def test_get_error_log():
-    handle = nvjitlink.create(1, ["-arch=sm_90"])
-    nvjitlink.complete(handle)
-    log_size = nvjitlink.get_error_log_size(handle)
-    log = bytearray(log_size)
-    nvjitlink.get_error_log(handle, log)
-    assert len(log) == log_size
-    nvjitlink.destroy(handle)
+    for option in ARCHITECTURES:
+        handle = nvjitlink.create(1, [f"-arch={option}"])
+        nvjitlink.complete(handle)
+        log_size = nvjitlink.get_error_log_size(handle)
+        log = bytearray(log_size)
+        nvjitlink.get_error_log(handle, log)
+        assert len(log) == log_size
+        nvjitlink.destroy(handle)
 
 
 def test_get_info_log():
-    handle = nvjitlink.create(1, ["-arch=sm_90"])
-    nvjitlink.add_data(handle, nvjitlink.InputType.ANY, ptx_kernel_bytes, len(ptx_kernel_bytes), "test_data")
-    nvjitlink.complete(handle)
-    log_size = nvjitlink.get_info_log_size(handle)
-    log = bytearray(log_size)
-    nvjitlink.get_info_log(handle, log)
-    assert len(log) == log_size
-    nvjitlink.destroy(handle)
+    for option, ptx_bytes in zip(ARCHITECTURES, ptx_kernel_bytes):
+        handle = nvjitlink.create(1, [f"-arch={option}"])
+        nvjitlink.add_data(handle, nvjitlink.InputType.ANY, ptx_bytes, len(ptx_bytes), "test_data")
+        nvjitlink.complete(handle)
+        log_size = nvjitlink.get_info_log_size(handle)
+        log = bytearray(log_size)
+        nvjitlink.get_info_log(handle, log)
+        assert len(log) == log_size
+        nvjitlink.destroy(handle)
 
 
 def test_get_linked_cubin():
-    handle = nvjitlink.create(1, ["-arch=sm_90"])
-    nvjitlink.add_data(handle, nvjitlink.InputType.ANY, ptx_kernel_bytes, len(ptx_kernel_bytes), "test_data")
-    nvjitlink.complete(handle)
-    cubin_size = nvjitlink.get_linked_cubin_size(handle)
-    cubin = bytearray(cubin_size)
-    nvjitlink.get_linked_cubin(handle, cubin)
-    assert len(cubin) == cubin_size
-    nvjitlink.destroy(handle)
+    for option, ptx_bytes in zip(ARCHITECTURES, ptx_kernel_bytes):
+        handle = nvjitlink.create(1, [f"-arch={option}"])
+        nvjitlink.add_data(handle, nvjitlink.InputType.ANY, ptx_bytes, len(ptx_bytes), "test_data")
+        nvjitlink.complete(handle)
+        cubin_size = nvjitlink.get_linked_cubin_size(handle)
+        cubin = bytearray(cubin_size)
+        nvjitlink.get_linked_cubin(handle, cubin)
+        assert len(cubin) == cubin_size
+        nvjitlink.destroy(handle)
 
 
 def test_get_linked_ptx():

```

Comment:
pass the suggested fixture here to get `empty_kernel_ltoir`
```suggestion
def test_get_linked_ptx(get_dummy_ltoir):
```

Reply:
ditto